src/matrix_t/matrix_t.h

```diff
@@ -66,6 +69,38 @@ class matrix_t : public data_t {
     channel = get_channel(data_type) | 0;
     size = (cols * channel) * rows;
   };
+  matrix_t(matrix_t &m) {
```

Comment:
this should let copy all properties and also the data array. The `val` (emscripten) data will not copied but we need thys copy constructor to be used internally in the C++ code.

Reply:
I will think about this. I have not implemented the `CopyTo` function, honestly i'm not sure how it should be done.